generator/src/main/java/org/apache/kafka/message/MessageGenerator.java

```diff
@@ -169,20 +185,20 @@ public static void processDirectories(String packageName, String outputDir, Stri
                         generator.write(writer);
                     }
                     numProcessed++;
-                    messageTypeGenerator.registerMessageType(spec);
+                    if (typeClassGenerator != null) {
+                        typeClassGenerator.registerMessageType(spec);
+                    }
                 } catch (Exception e) {
                     throw new RuntimeException("Exception while processing " + inputPath.toString(), e);
                 }
             }
         }
-        if (messageTypeGenerator.hasRegisteredTypes()) {
-            Path factoryOutputPath = Paths.get(outputDir, API_MESSAGE_TYPE_JAVA);
-            outputFileNames.add(API_MESSAGE_TYPE_JAVA);
+        if (typeClassGenerator != null) {
```

Comment:
Instead of using null ptr, maybe we could get a `NoOpTypeClassGenerator` which will do nothing for `generateAndWrite`, and remove the `outputName()` to make one single API. Something like
```
String outputFileName = typeClassGenerator.generateAndWrite(outputDir, writer);
```
I don't feel strong about this change, ideally it should look better, but up to you.

Reply:
I thought about that, but it doesn't really make anything simpler.  If there's no typeclass, we don't want to add an output file, we don't want to write to it, etc. etc.  So we still need all the same logic.